runtime/lua/vim/lsp/buf.lua

```diff
@@ -111,6 +111,39 @@ function M.completion(context)
   return request('textDocument/completion', params)
 end
 
+--@private
+--- If there is more than one client with formatting capability, asks the user
+--- which one to use.
+--
+--@returns The client to use for formatting
+local function get_formatting_client()
+  local clients = vim.tbl_values(vim.lsp.buf_get_clients());
+  clients = vim.tbl_filter(function (client)
+    return client.resolved_capabilities.document_formatting
+  end, clients)
+  -- better UX when choices are always in the same order (between restarts)
+  table.sort(clients, function (a, b) return a.name < b.name end)
+
+  if #clients > 1 then
+    local choices = {}
+    for k,v in ipairs(clients) do
+      table.insert(choices, string.format("%d %s", k, v.name))
+    end
+    local user_choice = vim.fn.confirm(
+      "Multiple formatters available, which one would you like to use?",
```

Comment:
This sounds a bit too wordy to me (nitpick). Maybe, "Select a language server for formatting:"

Reply:
Agreed